RecoHI/HiTracking/plugins/HIPixelTracksSlimmer_miniAOD.cc

```diff
@@ -0,0 +1,256 @@
+
+  void addPackedCandidate(std::vector<pat::PackedCandidate>& cands,
+                          const reco::TrackRef& trk,
+                          const reco::VertexRef& pvSlimmed,
+                          const reco::VertexRefProd& pvSlimmedColl,
+                          bool passPixelTrackSel) const;
+
+private:
+  void beginStream(edm::StreamID) override;
+  void produce(edm::Event&, const edm::EventSetup&) override;
+  void endStream() override;
+
+  //virtual void beginRun(edm::Run const&, edm::EventSetup const&) override;
+  //virtual void endRun(edm::Run const&, edm::EventSetup const&) override;
+  //virtual void beginLuminosityBlock(edm::LuminosityBlock const&, edm::EventSetup const&) override;
+  //virtual void endLuminosityBlock(edm::LuminosityBlock const&, edm::EventSetup const&) override;
+
+  // ----------member data ---------------------------
+  const edm::EDGetTokenT<reco::TrackCollection> srcTracks_;
+  const edm::EDGetTokenT<reco::VertexCollection> srcVertices_;
+  const edm::EDGetTokenT<reco::VertexCollection> srcPrimaryVertices_;
+  const double dzSigCut_;
+  const double dxySigCut_;
+  const double dzSigHP_;
+  const double dxySigHP_;
+  const double ptMax_;
+  const int covarianceVersion_;
+  const int covarianceSchema_;
+};
+
+//
+// constants, enums and typedefs
+//
+
+//
+// static data member definitions
+//
+
+//
+// constructors and destructor
+//
+HIPixelTracksSlimmer_miniAOD::HIPixelTracksSlimmer_miniAOD(const edm::ParameterSet& iConfig)
+    : srcTracks_(consumes<reco::TrackCollection>(iConfig.getParameter<edm::InputTag>("srcTracks"))),
+      srcVertices_(consumes<reco::VertexCollection>(iConfig.getParameter<edm::InputTag>("srcVertices"))),
+      srcPrimaryVertices_(consumes<reco::VertexCollection>(iConfig.getParameter<edm::InputTag>("srcPrimaryVertices"))),
+      dzSigCut_(iConfig.getParameter<double>("dzSigCut")),
+      dxySigCut_(iConfig.getParameter<double>("dxySigCut")),
+      dzSigHP_(iConfig.getParameter<double>("dzSigHP")),
+      dxySigHP_(iConfig.getParameter<double>("dxySigHP")),
+      ptMax_(iConfig.getParameter<double>("ptMax")),
+      covarianceVersion_(iConfig.getParameter<int>("covarianceVersion")),
+      covarianceSchema_(iConfig.getParameter<int>("covarianceSchema")) {
+  produces<std::vector<reco::Track>>();
+  produces<std::vector<pat::PackedCandidate>>();
+  produces<edm::Association<pat::PackedCandidateCollection>>();
+}
+
+HIPixelTracksSlimmer_miniAOD::~HIPixelTracksSlimmer_miniAOD() {
+  // do anything here that needs to be done at destruction time
+  // (e.g. close files, deallocate resources etc.)
+}
+
+//
+// member functions
+//
+
+// ------------ method called to produce the data  ------------
+void HIPixelTracksSlimmer_miniAOD::produce(edm::Event& iEvent, const edm::EventSetup& iSetup) {
+  using namespace edm;
+  using namespace std;
+
+  //track collection
+  edm::Handle<reco::TrackCollection> tracks;
+  iEvent.getByToken(srcTracks_, tracks);
```

Comment:
```suggestion
  auto tracks = iEvent.getHandle(srcTracks_);
```
is more compact
same below

Reply:
This comment is included in this commit : https://github.com/cms-sw/cmssw/pull/30313/commits/32f0d405268114c4305a6a71dc4b1d198c6d5ddf